src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -161,3 +173,113 @@ export const handleRedirect: RequestHandler<
     redirectPath,
   })
 }
+
+/**
+ * Handler for GET /:formId/publicform endpoint
+ * @returns 200 if the form exists
+ * @returns 404 if form with formId does not exist or is private
+ * @returns 410 if form has been archived
+ * @returns 500 if database error occurs or if the type of error is unknown
+ */
+export const handleGetPublicForm: RequestHandler<{ formId: string }> = async (
+  req,
+  res,
+) => {
+  const { formId } = req.params
+
+  const formResult = await getFormIfPublic(formId).andThen((form) =>
+    FormService.checkFormSubmissionLimitAndDeactivateForm(form),
+  )
+
+  // Early return if form is not public or any error occurred.
+  if (formResult.isErr()) {
```

Comment:
Should we only log with `logger.error` on database errors? Worried about non-errors being logged that desensitises us